src/idl_gen_cpp.cpp

```diff
@@ -1637,6 +1654,10 @@ class CppGenerator : public BaseGenerator {
     code_ += "};";  // End of table.
     code_ += "";
 
+    if (parser_.opts.generate_object_based_api) {
+        GenNativeTable(struct_def);
```

Comment:
can you run `generate_code.sh` to see if your changes have any effect on the current code? Moving where code is generated would show such changes. It's best if you can keep such changes to a minimum

Reply:
this has been done and the results commited.